replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/QueryServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package org.codice.ditto.replication.api.impl;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.stream.Stream;
+import org.apache.commons.io.IOUtils;
+import org.codice.ditto.replication.api.QueryService;
+import org.codice.ditto.replication.api.ReplicationException;
+import org.codice.ditto.replication.api.data.Query;
+import org.opengis.filter.Filter;
+
+/** A service which fetches query metacards from the catalog and returns them as {@link Query}s. */
+public class QueryServiceImpl implements QueryService {
+
+  private final CatalogFramework framework;
+
+  private final FilterBuilder builder;
+
+  private final InputTransformer xmlTransformer;
+
+  /**
+   * Creates a new QueryService
+   *
+   * @param framework The catalog framework to retrieve metacards from.
+   * @param builder A filter builder
+   * @param xmlTransformer An xml input transformer for transforming the query metacard xml on a
+   *     workspace metacard into a {@link Metacard} object.
+   */
+  public QueryServiceImpl(
+      CatalogFramework framework, FilterBuilder builder, InputTransformer xmlTransformer) {
+    this.framework = framework;
+    this.builder = builder;
+    this.xmlTransformer = xmlTransformer;
+  }
+
+  /**
+   * Provides a stream for retrieving all saved user queries as {@link Query}s.
+   *
+   * @return all the user queries saved on the system
+   */
+  @Override
+  public Stream<Query> queries() {
+    Filter filter = builder.attribute(Core.METACARD_TAGS).is().equalTo().text("workspace");
+    QueryImpl query = new QueryImpl(filter);
+    query.setPageSize(100);
+    QueryResponse queryResponse;
+    try {
+      queryResponse = framework.query(new QueryRequestImpl(query));
+    } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+      throw new ReplicationException("Failed to retrieve Query metacards.", e);
```

Comment:
❓ Should we really be using this exception for this?

Reply:
If not then we should declare it can be thrown

Reply:
And that would need to be done in the interface as we shouldn't be throwing out exceptions in an implementation that are not documented in the interface.

Reply:
I created a QueryException and it throws that now. I also added documentation to all the methods it could bubble out of.